lib/sdam/cursor.js

```diff
@@ -1,749 +0,0 @@
-'use strict';
```

Comment:
The big changes here were to refactor into class syntax, and clean up some of the private implementation. It turns out for very complicated reasons, that this is just not possible today because of the way our cursors do pseudo multiple inheritance. The `Cursor` class is exported and inherits `Readable` but only in native. Ideally, we would remove that dependency and have `cursor.stream()` return a specific stream instance but I want to punt that to future work.